### PR TITLE
gracefull parser abort when error detected during import

### PR DESCRIPTION
**Pull request by gaiaclary**
I found a sort of data error that would create bad behavior (null pointer exceptions) within the openCollada library functions. Actually the Collada file does not follow the Collada specifications in this case, see bugreport here: https://developer.blender.org/T50996

The following patch avoids nullpointer issues, allows to handle this sort of error in a more appropriate way via a callback to an error handler and a graceful abort of the import of the broken data.

### CODE REVIEW

File: COLLADASaxFrameworkLoader/src/COLLADASaxFWLMeshLoader.cpp [hunk 1]
```diff
@@ -913,11 +920,13 @@ namespace COLLADASaxFWL
...
             }
         }
+		return false;
     }
 
     //------------------------------
-    void MeshLoader::initializeTexCoordsOffset ()
+    bool MeshLoader::initializeTexCoordsOffset ()
     {
+		bool has_errors = false;
```
Comment (RemiArnaud) [inline COLLADASaxFrameworkLoader/src/COLLADASaxFWLMeshLoader.cpp]: not used ?

Comment (gaiaclary) [inline COLLADASaxFrameworkLoader/src/COLLADASaxFWLMeshLoader.cpp]: yes, not used. That was a left over that i actually never needed. I tried to change the pull request but i could not find out how to do that. Shall i close it and create a new one? Or if its otherwise OK, you could just remove that line?

Comment (RemiArnaud) [inline COLLADASaxFrameworkLoader/src/COLLADASaxFWLMeshLoader.cpp]: its quite simple, you just have to push changes in your own branch (gaiaclary:improved_parser_abort_on_error) , the one referred by this PR, and it will automatically be updated.

File: COLLADASaxFrameworkLoader/src/COLLADASaxFWLMeshLoader.cpp [hunk 2]
```diff
@@ -892,8 +900,7 @@ namespace COLLADASaxFWL
                 SourceBase* sourceBase = getSourceById ( sourceId );
                 if ( sourceBase == 0 ) 
                 {
-                    handleFWLError ( SaxFWLError::ERROR_DATA_NOT_VALID, "SourceBase of tex coords with semantic TEXCOORD not valid!" );
-                    return;
+                    return handleFWLError ( SaxFWLError::ERROR_DATA_NOT_VALID, "SourceBase of tex coords with semantic TEXCOORD not valid!" );
```
Comment (mbarnes-sb) [inline COLLADASaxFrameworkLoader/src/COLLADASaxFWLMeshLoader.cpp]: This message is a mistaken for this function. Looks like a copy& paste error

File: COLLADASaxFrameworkLoader/src/COLLADASaxFWLMeshLoader.cpp [hunk 3]
```diff
@@ -876,7 +884,7 @@ namespace COLLADASaxFWL
     }
 
     //------------------------------
-    void MeshLoader::initializeColorsOffset ()
+    bool MeshLoader::initializeColorsOffset ()
```
Comment (mbarnes-sb) [inline COLLADASaxFrameworkLoader/src/COLLADASaxFWLMeshLoader.cpp]: I that that color is as optional as normals, tangents, binormals. I don't see why this is coded as required data.

File: COLLADASaxFrameworkLoader/src/COLLADASaxFWLMeshLoader.cpp [hunk 4]
```diff
@@ -913,11 +920,13 @@ namespace COLLADASaxFWL
                 mColorList.push_back ( color );
             }
         }
+		return false;
     }
 
     //------------------------------
-    void MeshLoader::initializeTexCoordsOffset ()
+    bool MeshLoader::initializeTexCoordsOffset ()
```
Comment (mbarnes-sb) [inline COLLADASaxFrameworkLoader/src/COLLADASaxFWLMeshLoader.cpp]: Strictly speaking, texcoords are as optional as normals, etc.. The only data that is required is positions. A mesh without texcoords that has a bound texture could have texcoords generated (or even texture ignored) by the importer.

File: COLLADASaxFrameworkLoader/src/COLLADASaxFWLMeshLoader.cpp (hunk 2 above)
Comment (gaiaclary) [inline COLLADASaxFrameworkLoader/src/COLLADASaxFWLMeshLoader.cpp]: Could be, but wouldn't fixing the message be actually another patch? Unrelated to this pull request?

File: COLLADASaxFrameworkLoader/src/COLLADASaxFWLMeshLoader.cpp (hunk 3 above)
Comment (gaiaclary) [inline COLLADASaxFrameworkLoader/src/COLLADASaxFWLMeshLoader.cpp]: nevertheless the function internally calls the application error handler. But when the application error handler returns "abort parsing" then this function should not silently ignore this request, but give this information back to the caller.

Imho the correct fix would be to avoid calling the application error handler from within initializeColorsOffset () but i can not judge if that is right or wrong, so i guess its better to make another patch for fixing that part? 

as a sidenote, from my work with blender i learned to make many small commits rather than collections of loosely related fixes...

File: COLLADASaxFrameworkLoader/src/COLLADASaxFWLMeshLoader.cpp (hunk 4 above)
Comment (gaiaclary) [inline COLLADASaxFrameworkLoader/src/COLLADASaxFWLMeshLoader.cpp]: same as above, the function calls the application error handler, so either avoid that call or report back the error handler's decision for continue parsing or abort.

**Comment by RemiArnaud** [comment]
Thanks a lot for your contribution !
